### DISCUSSION TITLE
Browser HTML preview?

### DISCUSSION THREAD

**Discussion by datawench** [mpdf/mpdf#1680] [Q&A]
If I were to build an HTML file which I knew would be the actual content supplied to `Mpdf->WriteHTML()`, how would I preview it in such a way that I could be confident that it would match the actual look of the PDF generated by `Mpdf->Output()`? In other words, what, if any, browser model does Mpdf use to layout and style its output? I would have thought that perhaps headless Chrome might do it, but it does not appear to.

The motivation for this is that I want to be able to work on the layout directly in HTML, and have a good idea what it's going to look like when the PDF is finally created, rather than have to regenerate the PDF every time I make a change.

**Comment by finwe** [comment]
You would have to actually generate the PDF, convert it to an image and display the image as a preview.

To accurately mirror HTML/CSS in PDF, it is best to use headless Chrome, then the preview in HTML will be the same as the generated PDF.
